tests/expr_and_series/pipe_test.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any
+
+import narwhals as nw
+from tests.utils import compare_dicts
+
+
+@nw.narwhalify()
+def func(s: Any) -> Any:
+    return s.pipe(lambda x: x**2)
+
+
+input_list = {"a": [2, 4, 6, 8]}
+expected = [i**2 for i in input_list["a"]]
```

Comment:
can you construct `expected` explicitly please? else we risk having a bug in both paths and it going unnoticed

Reply:
this comment hasn't been addressed, please don't click 'resolve' until it has

Reply:
I just haven't pushed the commit, but I am declaring it explicitly locally.